test/service/src/lib.rs

```diff
@@ -0,0 +1,550 @@
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+mod chain_spec;
+mod genesis;
+
+pub use chain_spec::*;
+pub use genesis::*;
+
+use ansi_term::Color;
+use core::future::Future;
+use cumulus_collator::CollatorBuilder;
+use cumulus_network::DelayedBlockAnnounceValidator;
+use cumulus_primitives::ParaId;
+use cumulus_service::{
+	prepare_node_config, start_full_node, StartCollatorParams, StartFullNodeParams,
+};
+use polkadot_primitives::v0::CollatorPair;
+use sc_client_api::execution_extensions::ExecutionStrategies;
+use sc_client_api::{Backend as BackendT, BlockBackend, Finalizer, UsageProvider};
+use sc_executor::native_executor_instance;
+pub use sc_executor::NativeExecutor;
+use sc_informant::OutputFormat;
+use sc_network::{config::TransportConfig, multiaddr, NetworkService};
+use sc_service::{
+	config::{
+		DatabaseConfig, KeystoreConfig, MultiaddrWithPeerId, NetworkConfiguration,
+		OffchainWorkerConfig, PruningMode, WasmExecutionMethod,
+	},
+	BasePath, ChainSpec, Configuration, Error as ServiceError, PartialComponents, Role,
+	RpcHandlers, TFullBackend, TFullClient, TaskExecutor, TaskManager,
+};
+use sp_api::ConstructRuntimeApi;
+use sp_blockchain::HeaderBackend;
+use sp_consensus::{BlockImport, Environment, Error as ConsensusError, Proposer};
+use sp_core::{crypto::Pair, H256};
+use sp_keyring::Sr25519Keyring;
+use sp_runtime::traits::{BlakeTwo256, Block as BlockT};
+use sp_state_machine::BasicExternalities;
+use sp_trie::PrefixedMemoryDB;
+use std::sync::Arc;
+use substrate_test_client::BlockchainEventsExt;
+use test_primitives::Block;
+
+// Native executor instance.
+native_executor_instance!(
+	pub RuntimeExecutor,
+	cumulus_test_runtime::api::dispatch,
+	cumulus_test_runtime::native_version,
+);
+
+/// Starts a `ServiceBuilder` for a full service.
+///
+/// Use this macro if you don't actually need the full service, but just the builder in order to
+/// be able to perform chain operations.
+pub fn new_partial<RuntimeApi, Executor>(
+	config: &mut Configuration,
+) -> Result<
+	PartialComponents<
+		TFullClient<Block, RuntimeApi, Executor>,
+		TFullBackend<Block>,
+		(),
+		sp_consensus::import_queue::BasicQueue<Block, PrefixedMemoryDB<BlakeTwo256>>,
+		sc_transaction_pool::FullPool<Block, TFullClient<Block, RuntimeApi, Executor>>,
+		(),
+	>,
+	sc_service::Error,
+>
+where
+	RuntimeApi: ConstructRuntimeApi<Block, TFullClient<Block, RuntimeApi, Executor>>
+		+ Send
+		+ Sync
+		+ 'static,
+	RuntimeApi::RuntimeApi: sp_transaction_pool::runtime_api::TaggedTransactionQueue<Block>
+		+ sp_api::Metadata<Block>
+		+ sp_session::SessionKeys<Block>
+		+ sp_api::ApiExt<
+			Block,
+			Error = sp_blockchain::Error,
+			StateBackend = sc_client_api::StateBackendFor<TFullBackend<Block>, Block>,
+		> + sp_offchain::OffchainWorkerApi<Block>
+		+ sp_block_builder::BlockBuilder<Block>,
```

Comment:
Why is that soo generic?

Reply:
OK
![fire](https://media.giphy.com/media/JjiieDMHZ6pEI/giphy.gif)

Reply:
Mostly fixed in d2e8715e8add209c4cae9ea30d80a40ebcee97a7